.circleci/config.yml

```diff
@@ -0,0 +1,145 @@
+    steps:
+      - checkout
+      - attach_workspace:
+          at: .
+      - run: mkdir -p $TEST_RESULTS_DIR/hil
+      - run-gotests:
+          cmd: "gotestsum"
+          platform: "linux"
+
+      - persist_to_workspace:
+          root: .
+          paths:
+            - linux_cov_*.part
+
+      - store_test_results:
+          path: *TEST_RESULTS_DIR
+      - store_artifacts:
+          path: *TEST_RESULTS_DIR
+
+  win-tests:
+    executor: 
+      name: win/default
+      shell: bash --login -eo pipefail
+    environment:
+      <<: *ENVIRONMENT
+    working_directory: c:\gopath\src\github.com\hashicorp\hil
+    parameters:
+      go-version:
+        type: string
+      gotestsum-version:
+        type: string
+    steps: 
+      - run: git config --global core.autocrlf false
+      - checkout
+      - attach_workspace:
+          at: .
+      - run: 
+          name: Setup (remove pre-installed go v1.12)
+          command: |
+            rm -rf "c:\Go"
+            mkdir -p $TEST_RESULTS_DIR/hil
+
+      - restore_cache:
+          keys:
+            - win-golang-<< parameters.go-version >>-cache-v1
+            - win-gomod-cache-{{ checksum "go.mod" }}-v1
+
+      - run: 
+          name: Install go version << parameters.go-version >>
+          command: | 
+            if [ ! -d "c:\go" ]; then
+              echo "Cache not found, installing new version of go"
+              curl --fail --location https://dl.google.com/go/go<< parameters.go-version >>.windows-amd64.zip --output go.zip
+              unzip go.zip -d "/c"
+            fi
+
+      - run: 
+          name: Go mod download
+          command: |
+            go mod verify
+            go mod download
+  
+      - save_cache:
+          key: win-golang-<< parameters.go-version >>-cache-v1
+          paths:
+            - /go
+
+      - save_cache:
+          key: win-gomod-cache-{{ checksum "go.mod" }}-v1
+          paths:
+            - c:\Windows\system32\config\systemprofile\go\pkg\mod
+
+      - run:
+          name: Install gotestsum
+          command: |
+            curl --fail --location https://github.com/gotestyourself/gotestsum/releases/download/v<< parameters.gotestsum-version >>/gotestsum_<< parameters.gotestsum-version >>_windows_amd64.tar.gz --output gotestsum.tar.gz
+            tar -xvzf gotestsum.tar.gz
+
+      - run-gotests:
+          cmd: "./gotestsum.exe"
+          platform: "win"
+
+      - persist_to_workspace:
+          root: .
+          paths:
+            - win_cov_*.part
+
+      - store_test_results:
+          path: *WIN_TEST_RESULTS
+      - store_artifacts:
+          path: *WIN_TEST_RESULTS
+          
+workflows:
+  hil:
+    jobs:
+      - linux-tests:
+          matrix:
+            parameters:
+              go-version: ["1.14"]
+          name: linux-test-go-<< matrix.go-version >>
+      - win-tests:
+          matrix:
+            parameters:
+              go-version: ["1.14"]
+              gotestsum-version: ["0.4.1"]
```

Comment:
It feels a little weird that `gotestsum` version is configurable for Windows but not Linux, but not a big deal.

Reply:
on linux it's already pre-installed, but it's not on the windows executor. I could add a step to remove the current version and install the parameterized version on linux, but :shrug:

Reply:
My thought was more, "hardcode the version into the Windows version" not make the Linux version configurable. But, yeah, no strong opinion here, this is fine as is.